x-pack/plugins/maps/server/mvt/get_tile.ts

```diff
@@ -100,120 +109,135 @@ export async function getGridTile({
 
 export async function getTile({
   logger,
-  callElasticsearch,
+  context,
   index,
   geometryFieldName,
   x,
   y,
   z,
   requestBody = {},
   geoFieldType,
+  searchSessionId,
 }: {
   x: number;
   y: number;
   z: number;
   geometryFieldName: string;
   index: string;
-  callElasticsearch: (type: string, ...args: any[]) => Promise<unknown>;
+  context: RequestHandlerContext;
   logger: Logger;
   requestBody: any;
   geoFieldType: ES_GEO_FIELD_TYPE;
+  searchSessionId?: string;
 }): Promise<Buffer | null> {
-  const geojsonBbox = tileToGeoJsonPolygon(x, y, z);
-
-  let resultFeatures: Feature[];
+  let features: Feature[];
   try {
-    let result;
-    try {
-      const geoShapeFilter = {
-        geo_shape: {
-          [geometryFieldName]: {
-            shape: geojsonBbox,
-            relation: 'INTERSECTS',
-          },
+    requestBody.query.bool.filter.push({
+      geo_shape: {
+        [geometryFieldName]: {
+          shape: tileToGeoJsonPolygon(x, y, z),
+          relation: 'INTERSECTS',
         },
-      };
-      requestBody.query.bool.filter.push(geoShapeFilter);
+      },
+    });
 
-      const esSearchQuery = {
-        index,
-        body: requestBody,
-      };
+    const searchOptions = {
+      sessionId: searchSessionId,
+    };
 
-      const esCountQuery = {
-        index,
-        body: {
-          query: requestBody.query,
+    const countResponse = await context
```

Comment:
Change count from using `count` API to using `search` API. This is because needed to use `search` API to pass searchSessionId. There is no performance degradation because under the covers, elasticsearch `count` API is just a wrapper for `search` API

Reply:
great findm thx